python/lsst/dbb/buffmngrs/endpoint/finder/finder.py

```diff
@@ -295,26 +304,61 @@ def parse_rsync_logs(directory, blacklist=None, isodate=None, timespan=1):
     """
     if blacklist is None:
         blacklist = []
-    end = datetime.date.today()
+    delay = timedelta(seconds=delay)
+    end = date.today()
     if isodate is not None:
         end = isodate
-    dates = [end - datetime.timedelta(days=n) for n in range(timespan+1)]
-    for date in dates:
-        top = os.path.join(directory, date.isoformat().replace("-", ""))
+    dates = [end - timedelta(days=n) for n in range(timespan+1)]
+    for day in dates:
+        top = os.path.join(directory, day.isoformat().replace("-", ""))
         if not os.path.exists(top):
             continue
-        for dirpath, dirnames, filenames in os.walk(top):
+        for dirpath, _, filenames in os.walk(top):
             manifests = [os.path.join(dirpath, fn) for fn in filenames
-                         if re.match(r"^rsync.*log$", fn)]
+                         if re.match(r"rsync.*log$", fn)]
             for manifest in manifests:
-                if os.path.exists(manifest + ".done"):
+                sentinel = ".".join([manifest, extension])
+
+                # Ignore a log file if it was modified withing a specified
+                # time span, i.e., assume that its transfer has not finished
+                # yet.
+                manifest_mtime = None
+                try:
+                    status = os.stat(manifest)
+                except FileNotFoundError:
+                    logger.error(f"{manifest}: file not found")
+                else:
+                    manifest_mtime = datetime.fromtimestamp(status.st_mtime)
+                now = datetime.now()
+                if manifest_mtime is None or now - manifest_mtime < delay:
                     continue
+
+                # Check if a log file was already parsed, i.e., a sentinel
+                # file exists. If it wasn't (no sentinel), do nothing; file
+                # hasn't been parsed yet. If it was parsed, check if by
+                # chance the log file hasn't been modified AFTER it was
+                # marked as parsed. It that's the case, remove the sentinel.
```

Comment:
When removing the sentinel file and later reparsing the log file for "new filenames", please make sure messages are clear between this case an other duplicate file cases.